apps/sms/js/activity_handler.js

```diff
@@ -125,21 +115,31 @@ var ActivityHandler = {
       threadId: null
     };
 
-    // try to get a thread from number
-    // if no thread, promise is rejected and we try to find a contact
-    return MessageManager.findThreadFromNumber(viewInfo.number).then(
-      function onResolve(threadId) {
-        viewInfo.threadId = threadId;
-      },
-      function onReject() {
-        return ActivityHandler._findContactByNumber(viewInfo.number)
-          .then( (contact) => viewInfo.contact = contact);
-      }
-    )
-    // case no contact and no thread id: gobble the error
-    .catch(() => {})
-    // finally call toView whatever contact and threadId we have.
-    .then( () => this.toView(viewInfo) );
+    var viewInfoPromise;
+    if (viewInfo.number) {
+      // try to get a thread from number
+      // if no thread, promise is rejected and we try to find a contact
+      viewInfoPromise = MessageManager.findThreadFromNumber(viewInfo.number).
+        then(function onResolve(threadId) {
+          viewInfo.threadId = threadId;
+        }, function onReject() {
+          return ActivityHandler._findContactByNumber(viewInfo.number)
+            .then((contact) => viewInfo.contact = contact);
+        }
+      )
+      // case no contact and no thread id: gobble the error
+      .catch(() => {});
+    }
+
+    return (viewInfoPromise || Promise.resolve()).
+      then(() => this.toView(viewInfo)).
+      then(() => {
+        // Focus on Compose only in case we navigated to Thread panel, as for
+        // Compose all focus-related stuff is handled by ThreadUI itself
```

Comment:
Sure we can, I had this in mind too, just tell me what approach you like more :)